### PR TITLE
Login: Show a notice when we send the SMS code

### PR DESCRIPTION
**Pull request by scruffian**
From #14173:

> Should we display the notice that confirms a code has been sent when a user has 2FA via SMS enabled?

This PR enables this message in the case where the SMS gets sent as the initial 2FA method:
  
#### Testing instructions
  
1. Run `git checkout add/2fa-sms-message` and start your server
2. Apply the patch D6404-code
2. Open the [`Login` page](http://calypso.localhost:3000/log-in)
3. Log in to a 2FA account that uses SMS as their primary form of authentication
3. Check that the success notice shows
  
  
#### Reviews
  
- [x] Code
- [x] Product

### CODE REVIEW

File: client/state/login/actions.js [hunk 1]
```diff
@@ -150,6 +159,19 @@ export const loginUser = ( usernameOrEmail, password, rememberMe, redirectTo ) =
 				rememberMe,
 				data: response.body && response.body.data,
 			} );
+
+			if ( response.body.data.two_step_notification_sent === 'sms' ) {
+				const message = getSMSMessageFromResponse( response );
+				setTimeout( function() {
```
Comment (scruffian) [inline client/state/login/actions.js]: We should do this another way, but I'm not sure how.

**Comment by fabianapsimoes** [comment]
Looks good to me, but should it say "Text message sent to phone number ending in 1111" instead of "Message sent to phone number ending in 1111"? That would make it more consistent with the terminology @ranh proposed earlier.

**Comment by yurynix** [comment]
The `setTimeout` is required because of the redirect [here](https://github.com/Automattic/wp-calypso/blob/master/client/blocks/login/index.jsx#L63) that is called from [here](https://github.com/Automattic/wp-calypso/blob/master/client/blocks/login/login-form.jsx#L91)

That triggers the action:
screen shot 2017-07-17 at 20 23 35

which clears the notice reducer.

the problem is that the redirect happens in a promise handler _after_ we set the notice 😭 
I don't know how to solve that more elegantly than https://github.com/Automattic/wp-calypso/pull/16233/commits/6004b0e81beef9302023644779d11f2cab6379af which isn't an improvement on the `setTimeout` so feel free to drop that commit.

**Comment by scruffian** [comment]
@yurynix, I like your solution better :)

**Comment by scruffian** [comment]
Code 👍 
Product 👍

File: client/state/login/actions.js [hunk 2]
```diff
@@ -150,6 +159,16 @@ export const loginUser = ( usernameOrEmail, password, rememberMe, redirectTo ) =
...
 				data: response.body && response.body.data,
 			} );
+
+			if ( get( response, 'body.data.two_step_notification_sent', null ) === 'sms' ) {
+				dispatch( {
+					type: TWO_FACTOR_AUTHENTICATION_SEND_SMS_CODE_REQUEST_SUCCESS,
+					notice: {
+						message: getSMSMessageFromResponse( response ),
+						status: 'is-success'
+					},
```
Comment (stephanethomas) [inline client/state/login/actions.js]: We're missing the `two_step_nonce_sms` nonce here.

**Pull request merged**